### PR TITLE
fix: add Gemini model's pricing and token count

### LINKED ISSUE #744: Gemini model's pricing is input & output separated, but only got total token count
image

image

### PR DESCRIPTION
**Pull request by TravisRoad**
Fix: #744

### CODE REVIEW

**Review by TravisRoad — COMMENTED** [review]
Is it necessary to integrate question, prompt, and knowledge to generate in `QueryText` function? The previous `QueryText` function did not implement this logic.

File: model/gemini.go [hunk 1]
```diff
@@ -59,14 +60,14 @@ func (p *GeminiModelProvider) calculatePrice(modelResult *ModelResult) error {
 		modelResult.ResponseTokenCount = modelResult.TotalTokenCount
 	}
 
+	model := p.subType
```
Comment (hsluoyz) [inline model/gemini.go]: Don't define model

File: model/gemini.go [hunk 2]
```diff
@@ -59,14 +60,14 @@ func (p *GeminiModelProvider) calculatePrice(modelResult *ModelResult) error {
...
 	var inputPricePerThousandTokens, outputPricePerThousandTokens float64
-	switch p.subType {
-	case "Gemini Pro":
+	switch {
+	case strings.Contains(model, "gemini-pro"):
```
Comment (hsluoyz) [inline model/gemini.go]: other models? only pro?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: model/gemini.go (hunk 2 above)
Comment (TravisRoad) [reply model/gemini.go]: there is only gemini pro model for now.
But there are legacy PaLM models. I'll add lines here.

Comment (hsluoyz) [reply model/gemini.go]: No need to add those, thanks

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**